integration/renamecollection_compat_test.go

```diff
@@ -34,140 +34,179 @@ func TestRenameCollectionCompat(t *testing.T) {
 		AddNonExistentCollection: true,
 	})
 
+	ctx, targetCollection, compatCollection := s.Ctx, s.TargetCollections[0], s.CompatCollections[0]
+
+	targetDB := targetCollection.Database()
+	compatDB := compatCollection.Database()
+
+	require.Equal(t, compatDB.Name(), targetDB.Name())
+	dbName := targetDB.Name()
+
+	require.Equal(t, compatCollection.Name(), targetCollection.Name())
+	cName := targetCollection.Name()
+
+	to := dbName + ".newCollection"
+	from := dbName + "." + cName
+
+	targetDBConnect := targetCollection.Database().Client().Database("admin")
+	compatDBConnect := compatCollection.Database().Client().Database("admin")
+
+	var targetRes bson.D
+	targetCommand := bson.D{{"renameCollection", from}, {"to", to}}
+	targetErr := targetDBConnect.RunCommand(ctx, targetCommand).Decode(&targetRes)
+	require.NoError(t, targetErr, "compat error; target returned no error")
+
+	var compatRes bson.D
+	compatCommand := bson.D{{"renameCollection", from}, {"to", to}}
+	compatErr := compatDBConnect.RunCommand(ctx, compatCommand).Decode(&compatRes)
+	require.NoError(t, compatErr, "compat error; target returned no error")
+
+	// Collection lists after rename must be the same
+	targetNames, err := targetDB.ListCollectionNames(ctx, bson.D{})
+	require.NoError(t, err)
+
+	compatNames, err := compatDB.ListCollectionNames(ctx, bson.D{})
+	require.NoError(t, err)
+
+	assert.ElementsMatch(t, targetNames, compatNames)
+
+	// Recreation of collection with the old name must be possible
+	err = targetDB.CreateCollection(ctx, cName)
+	require.NoError(t, err)
+
+	err = compatDB.CreateCollection(ctx, cName)
+	require.NoError(t, err)
+
+	// Collection lists after recreation must be the same
+	targetNames, err = targetDB.ListCollectionNames(ctx, bson.D{})
+	require.NoError(t, err)
+
+	compatNames, err = compatDB.ListCollectionNames(ctx, bson.D{})
+	require.NoError(t, err)
+
+	assert.ElementsMatch(t, targetNames, compatNames)
+
+	// Rename one more time
+	targetCommand = bson.D{{"renameCollection", from}, {"to", to + "_new"}}
+	targetErr = targetDBConnect.RunCommand(ctx, targetCommand).Decode(&targetRes)
+	require.NoError(t, targetErr)
+
+	compatCommand = bson.D{{"renameCollection", from}, {"to", to + "_new"}}
+	compatErr = compatDBConnect.RunCommand(ctx, compatCommand).Decode(&compatRes)
+	require.NoError(t, compatErr)
+
```

Comment:
Don't we want to call `ListCollectionNames` here too?

Reply:
Yes let's call it